src/pdf.rs

```diff
@@ -1,21 +1,42 @@
-// Copyright 2015-2016, The Gtk-rs Project Developers.
+// Copyright 2018-2019, The Gtk-rs Project Developers.
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
-use std::ffi::CString;
+use std::mem;
+use std::ffi::{CStr, CString};
 use std::ops::Deref;
 use std::path::Path;
 use std::io;
 
 use ffi;
-use ::enums::PdfVersion;
+use ::enums::{PdfVersion, PdfMetadata};
 use surface::Surface;
 use support::{self, FromRawSurface};
 
 #[cfg(feature = "use_glib")]
 use glib::translate::*;
 
 
+pub fn get_versions() -> Vec<PdfVersion> {
+    let vers_slice = unsafe {
+        let mut vers_ptr: *mut ffi::cairo_pdf_version_t = mem::uninitialized();
+        let mut num_vers = 0;
+        ffi::cairo_pdf_get_versions(&mut vers_ptr as _, &mut num_vers as _);
```

Comment:
Might this return `0` or `NULL`?

Reply:
I don't know, the documentation says nothing on that and it doesn't say anything about the returned pointer memory management either. The implementation simply returns a pointer to an internal static array (see [here](https://github.molgen.mpg.de/git-mirror/cairo/blob/master/src/cairo-pdf-surface.c#L165)). To be in line with that, it would probably be better to return a `&'static str` instead of `String` here. (The same applies to the others.)

Reply:
Ok so can't be `NULL` then. Thanks for checking!

For making it a `&str` you need to count until the `\0`-terminator for each string and then use the corresponding functions, but that would indeed be nicer.